### PR TITLE
evm: Upgradeable contracts

### PR DESCRIPTION
**Pull request by shohamc1**

### CODE REVIEW

File: lib/ain-contracts/src/lib.rs [hunk 1]
```diff
@@ -103,20 +103,60 @@ pub struct FixedContract {
 }
 
 lazy_static::lazy_static! {
-    pub static ref INTRINSIC_CONTRACT: FixedContract = {
-        let bytecode = solc_artifact_bytecode_str!("dfi_intrinsics", "deployed_bytecode.json");
+    pub static ref IMPLEMENTATION_SLOT: H256 = H256::from_str("0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc").unwrap();
```
Comment (Jouzo) [inline lib/ain-contracts/src/lib.rs:106]: Can drop the unwrap and pass the [u8; 32] array here. Can then be lifted out of lazy_static

File: lib/ain-evm/src/evm.rs [hunk 2]
```diff
@@ -182,36 +188,90 @@ impl EVMServices {
...
+
+            // deploy DST20 implementation contract
+            let DeployContractInfo {
+                address,
+                storage,
+                bytecode,
+            } = dst20_implementation();
+            trace!("deploying {:x?} bytecode {:?}", address, bytecode);
+            executor.deploy_contract(address, bytecode, storage)?;
+            executor.commit();
```
Comment (cuongquangnam) [inline lib/ain-evm/src/evm.rs:269]: no pushing to all_transactions and receipts?

Comment (shohamc1) [reply lib/ain-evm/src/evm.rs:275]: Fixed: 7eab71b00ea429d82d815a621400772cc3aac8b1

File: lib/ain-contracts/src/lib.rs (hunk 1 above)
Comment (shohamc1) [reply lib/ain-contracts/src/lib.rs]: Fixed: 78ffbf4

File: lib/ain-contracts/src/lib.rs [hunk 3]
```diff
@@ -8,7 +8,10 @@ pub type Result<T> = std::result::Result<T, anyhow::Error>;
 
 pub const DST20_ADDR_PREFIX_BYTE: u8 = 0xff;
 pub const INTRINSICS_ADDR_PREFIX_BYTE: u8 = 0xdf;
-
+pub const IMPLEMENTATION_SLOT: H256 = H256([
```
Comment (prasannavl) [inline lib/ain-contracts/src/lib.rs:11]: Could you comment in the address here as well?

Comment (shohamc1) [reply lib/ain-contracts/src/lib.rs:11]: Done: dab2d85

File: test/functional/feature_evm_dst20.py [hunk 4]
```diff
@@ -937,7 +937,7 @@ def run_test(self):
 
         # Contract ABI
         self.abi = open(
```
Comment (prasannavl) [inline test/functional/feature_evm_dst20.py:939]: Can we leave a short comment to indicate the rationale to use dst20 facade as the abi, but v1 as the bytecode ?

Comment (shohamc1) [reply test/functional/feature_evm_dst20.py:940]: Done: d5d3839dbcac45af568d3df3fb28c1064504f32b

**Pull request merged**